src/v/datalake/schema_registry.h

```diff
@@ -73,7 +77,9 @@ using get_proto_offsets_result
 // Extract the schema id from a record's value. This simply extracts the id. It
 // does not validate that the schema exists in the Schema Registry.
 get_schema_id_result get_value_schema_id(iobuf& record);
-get_proto_offsets_result get_proto_offsets(iobuf& record);
+get_proto_offsets_result get_schema_proto_offsets(iobuf& record);
+result<proto_offsets_message_data, get_schema_error>
+get_proto_offsets(iobuf& record);
```

Comment:
nitpick: does this actually need to be exposed to the outside world?

Reply:
In this PR, no. I will use it in v_datalake in an upcoming PR though.

Reply:
gotcha